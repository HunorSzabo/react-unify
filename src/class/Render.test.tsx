import * as React from 'react';
import {create} from 'react-test-renderer';
import {prop, Render, state} from '../';

@Render((vm: Counter) =>
  <div>
    <p>Count: {vm.count}</p>
    <button onClick={() => vm.increment()}>Increment by {vm.amount}</button>
  </div>
)
class Counter extends React.PureComponent<{ amount?: number }> {
  @prop amount: number = 1;
  @state count: number = 0;

  increment() {
    this.count += this.amount;
  }
}

it('renders Counter component', () => {
  const renderer = create(<Counter/>);
  renderer.root.findByType(Counter).instance.increment();
  renderer.root.findByType('button').props.onClick();
  expect(renderer).toMatchSnapshot();
});
