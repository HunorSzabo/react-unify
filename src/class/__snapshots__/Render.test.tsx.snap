// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Counter component 1`] = `
<div>
  <p>
    Count: 
    2
  </p>
  <button
    onClick={[Function]}
  >
    Increment by 
    1
  </button>
</div>
`;
